src/main/java/sparksoniq/spark/DataFrameUtils.java

```diff
@@ -27,8 +27,12 @@
         StructType schema = generateSchemaFromSchemaItem(schemaItem, metadata);
         JavaRDD<Row> rowRDD = itemRDD.map((Function<Item, Row>) item -> ItemParser.getRowFromItem(item));
         try {
-            return SparkSessionManager.getInstance().getOrCreateSession().createDataFrame(rowRDD, schema);
-        } catch (IllegalArgumentException ex) {
+            Dataset<Row> result = SparkSessionManager.getInstance()
+                .getOrCreateSession()
+                .createDataFrame(rowRDD, schema);
+            result.take(1);
+            return result;
+        } catch (ClassCastException | IllegalArgumentException ex) {
```

Comment:
Bugfix with Annotate:
HybridRunTimeIterator::getRDD() method receives the dataframe result of annotate() calculated here. Then df.javaRDD() Spark action is called on the result which triggers a ClassCastException when it fails to cast doubles to integers(as specified by the schema). 
<img width="796" alt="image" src="https://user-images.githubusercontent.com/13052327/75806633-574dbc80-5d84-11ea-9f31-cd3cb6ccbc51.png">

I have added the take(1) action here to trigger the action and hence the exception earlier, so that we can give the nice error message as shown in the image above. It may not be ideal, but let me know if you agree with this solution or if you have a better alternative @ghislainfourny .

Reply:
This is fine for now (correctness first) and we can optimize later.